src/PeakFinder.cpp

```diff
@@ -0,0 +1,170 @@
+#include "PeakFinder.hpp"
+#include <algorithm>
+
+using namespace shasta;
+using std::sort;
+using std::cerr;
+
+
+void PeakFinder::findPeaks(vector<uint64_t>& y){
+    size_t xSize = y.size();
+
+    // Create a vector of values, in which each value can be used to fetch the parent peak for that position in the
+    // y. If that position does not yet belong to a peak, it has index -1
+    vector<int64_t> peakIndex(xSize, -1);
+
+    // Create a vector of x values which point to y values
+    vector<size_t> indexes(xSize, 0);
+    for (size_t i=0; i<indexes.size() ; i++) {
+        indexes[i] = i;
+    }
+
+    // Sort the x values based on their corresponding y
+    sort(indexes.begin(), indexes.end(),
```

Comment:
I find these lambda constructs unclear, but I can see that in this case the alternative would introduce additional complexity.